src/App.tsx

```diff
@@ -1,107 +1,114 @@
-import React from 'react';
-import flag from './pngegg.png'
-import logo from './logo.svg'
-
-import './App.css';
+import React from "react";
+import flag from "./pngegg.png";
+import "./App.css";
 
 function App() {
-  const iniImg = {1:flag, 2:flag, 3:flag, 4:flag,
-    5:flag, 6:flag, 7:flag, 8:flag, 9:flag, 10:flag, 11:flag, 12:flag, 13:flag, 14:flag, 15:flag,
-    16:flag, 17:flag, 18:flag, 19:flag, 20:flag, 21:flag};
-  const [img, setImg] = React.useState(iniImg);
+  const iniImg = new Array(21).fill(flag);
   const [flags, setFlags] = React.useState(21);
-  const [player, setPlayer] = React.useState('first');
+  const [player, setPlayer] = React.useState("first");
   const [took, setTook] = React.useState(0);
-  const [previousPlayer, setpreviousPlayer] = React.useState('first');
-
+  const [previousPlayer, setpreviousPlayer] = React.useState("first");
 
-  const onclick1 = () => {
-    setFlags(flags - 1);
-    setImg(prevState => ({
-      ...prevState,
-      [22-flags]: logo,
-    }))
-    if (flags === 1 || flags === 0) {
-      alert('Player ' + player + ' won' );
+  // it maps trough filtered array to make rows
+  // 'a' and 'b' changes the shape
+  const returnFlag = (a: number, b: number) => {
+    return (
+      <div key={"a"}>
+        {iniImg
+          .filter((elem, index) => index + a < b)
+          .map((item, i) => (
+            <img
+              key={i}
+              id={`${i + a}`}
+              className="flag"
+              src={item}
+              alt="simpleFlag"
+            />
+          ))}
+      </div>
+    );
+  };
+  // function that makes a pyramid out of flags
+  // it iterates returnFlag function 6 times and return array
+  // with returnFlag functions and also it changes a and b to get the shape of pyramid
+  const makeP = () => {
+    let a = 20;
+    let b = 21;
+    let tempA = 2;
+    let tempB = 1;
+    let f = [];
+    for (let i = 0; i < 6; i++) {
+      f.push(returnFlag(a, b));
+      a -= tempA;
+      b -= tempB;
+      tempA++;
+      tempB++;
     }
-    if (player === 'first') {
-      setpreviousPlayer('first');
+    return f;
+  };
+  // it changes the values of players, who took flags and which player turn it is
+  const checkPlayers = () => {
+    if (player === "first") {
+      setpreviousPlayer("first");
     } else {
-      setpreviousPlayer('second');
+      setpreviousPlayer("second");
     }
-    setTook(1)
-    if (player === 'first') {
-      setPlayer('second');
+    if (player === "first") {
```

Comment:
Because this if is the same as the first one, you can move setPlayer("second"); and setPlayer("first"); to the first if/else block and remove this one, to have less repetition in the code